### PR TITLE
Add functionalities

### PR DESCRIPTION
**Pull request by vsamy**
Hi,

I needed to add some more functionalities for me. 
It is up to you if you want to merge it.

Here are the modifications:
 - Add primal and dual feasibility tolerance.
 - Add warmstart option.
 - Add inform function that prints solver status.
 - Add possibility to hide/display solver output.
 - Correct some indents.

Vincent S.

### CODE REVIEW

File: src/Gurobi.cpp [hunk 1]
```diff
@@ -59,22 +59,134 @@ const VectorXd& GurobiCommon::result() const
...
 }
 
+int GurobiCommon::warmStart() const
+{
+	return model_.get(GRB_IntParam_MultiObjMethod);
+}
+
+void GurobiCommon::warmStart(int warmStatus)
+{
+	assert(tol >= -1 && tol <= 2);
```
Comment (haudren) [inline src/Gurobi.cpp]: This should be warmStatus. Note that asserts are only triggered in Debug mode.

File: src/Gurobi.cpp [hunk 2]
```diff
@@ -59,22 +59,134 @@ const VectorXd& GurobiCommon::result() const
 	return X_;
 }
 
+int GurobiCommon::warmStart() const
```
Comment (haudren) [inline src/Gurobi.cpp]: Would it be possible to return a more explicit warmStatus ? Some kind of enum like so maybe:
enum WarmStatus
{
     DEFAULT= -1,
     PRIMAL,
     DUAL,
     NONE
}

**Review by haudren — CHANGES_REQUESTED** [review]
Thanks for the improvements! If you could just change the warmStatus things, it'll be perfect :+1:

**Comment by vsamy** [comment]
The enum type for warmstatus is a great idea. I have added it and change the test to check the warmstart() functions. I deleted the assert it is no longer needed.

Note that i personnaly like adding assert to test my stuff in debug before going in release mode.

**Comment by haudren** [comment]
Nice! Thanks for the changes :)

I was not criticizing the fact that you use asserts, but rather that you should test your asserts in debug mode, so we agree!

**Review by haudren — APPROVED** [review]

**Pull request merged**